_inc/lib/class-jetpack-podcast-helper.php

```diff
@@ -65,6 +65,48 @@ public static function get_player_data( $feed ) {
 		return $player_data;
 	}
 
+	/**
+	 * Gets a specific track from the supplied feed URL.
+	 *
+	 * @param string $feed     The RSS feed to find the track in.
+	 * @param string $guid     The GUID of the track.
+	 * @return array|WP_Error  The track object or an error object.
+	 */
+	public static function get_track_data( $feed, $guid ) {
+		$feed = esc_url_raw( $feed );
+
+		// Try loading track data from the cache.
+		$transient_key = 'jetpack_podcast_' . md5( $feed ) . '_' . md5( $guid );
+		$track_data    = get_transient( $transient_key );
+
+		// Fetch data if we don't have any cached.
+		if ( false === $track_data || ( defined( 'WP_DEBUG' ) && WP_DEBUG ) ) {
+			// Load feed.
+			$rss = self::load_feed( $feed );
```

Comment:
@obenland did I catch it that you are trying to write unit tests which mock this `load_feed` call but it doesn't work because it's a static method?

If so could we try creating a new class in the test scope which extends the class under test? We would then inherit all the methods from the parent class (the one under test) but have the ability to stub out the `load_feed()` method simply by defining it on the child class we have created in the test scope. 

Some more on this here:

https://stackoverflow.com/a/47377592

Reply:
faed58b